velox/common/encode/Base64.h

```diff
@@ -71,62 +78,82 @@ class Base64 {
       std::string& output);
 
   /// Encodes the specified number of characters from the 'data' and writes the
-  /// result to the 'output'. The output must have enough space, e.g. as
-  /// returned by the calculateEncodedSize().
+  /// result to the 'output' using URL encoding. The output must have enough
+  /// space as returned by the calculateEncodedSize().
   static void encodeUrl(const char* data, size_t size, char* output);
 
-  // compatible with www's Base64URL::encode/decode
-  // TODO rename encode_url/decode_url to encodeUrl/encodeUrl.
+  /// Encodes the specified number of characters from the 'data' using URL
+  /// encoding.
   static std::string encodeUrl(const char* data, size_t len);
+
+  /// Encodes the specified IOBuf data using URL encoding.
   static std::string encodeUrl(const folly::IOBuf* data);
+
+  /// Encodes the specified text using URL encoding.
   static std::string encodeUrl(folly::StringPiece text);
+
+  /// Decodes the specified URL encoded payload and writes the result to the
+  /// 'output'.
   static void decodeUrl(
       const std::pair<const char*, int32_t>& payload,
       std::string& output);
+
+  /// Decodes the specified URL encoded text.
   static std::string decodeUrl(folly::StringPiece text);
 
+  /// Decodes the specified number of characters from the 'src' and writes the
+  /// result to the 'dst'.
   static size_t
   decode(const char* src, size_t src_len, char* dst, size_t dst_len);
 
+  /// Decodes the specified number of characters from the 'src' using URL
```

Comment:
It would be nice to add some links to material that explains the difference between 'encoding' and 'URL encoding'.

Would you clarify what happens if 'src' cannot be decoded?